domaincommon/src/state.rs

```diff
@@ -165,6 +184,38 @@ impl MatchState {
         state.turn_status.taken.insert(mech.to_string());
         state
     }
+
+    fn reduce_mech_actions(state: &MatchState, mech: &str, points_consumed: &u32) -> MatchState {
+        let mut state = state.clone();
+        let mech_state = state.mechs[mech].clone();
+        state.mechs.insert(
+            mech.to_string(),
+            MechState {
+                remaining_aps: mech_state.remaining_aps - points_consumed,
+                ..mech_state
+            },
+        );
+        state
+    }
+
+    fn reset_mech_actions(state: &MatchState) -> MatchState {
+        let mut state = state.clone();
+        state.mechs = state
+            .mechs
+            .clone()
```

Comment:
I'm not sure without a compiler at hand, but I don't think you need to clone it before you do `into_iter()` ... into_iter consumes the source of the iterable, and you're calling `collect` on the result of that consumption and then returning it (move). tl;dr I think we can remove this `clone` call.

Reply:
Just removing the clone call results in the compiler getting angry for borrow of moved value: `state`, playing around with it for a few minutes makes me believe it's necessary but I'm open to adding something else to make it work if it helps